fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -251,44 +324,50 @@ public boolean visit(java.lang.String elementName, int elementIndex, ContactPoin
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, HumanName humanName) {
         Parameter p;
+        if (!STRING.equals(searchParamType)) {
+            throw invalidComboException(searchParamType, humanName);
+        }
         if (humanName.getFamily() != null) {
             // family is just a string in R4 (not a list)
             p = new Parameter();
-            p.setName(name);
+            p.setName(searchParamCode);
             p.setValueString(humanName.getFamily().getValue());
             result.add(p);
         }
         for (com.ibm.fhir.model.type.String given : humanName.getGiven()) {
             p = new Parameter();
-            p.setName(name);
+            p.setName(searchParamCode);
             p.setValueString(given.getValue());
             result.add(p);
         }
         for (com.ibm.fhir.model.type.String prefix : humanName.getPrefix()) {
             p = new Parameter();
-            p.setName(name);
+            p.setName(searchParamCode);
             p.setValueString(prefix.getValue());
             result.add(p);
         }
         for (com.ibm.fhir.model.type.String suffix : humanName.getSuffix()) {
             p = new Parameter();
-            p.setName(name);
+            p.setName(searchParamCode);
             p.setValueString(suffix.getValue());
             result.add(p);
         }
         if (humanName.getText() != null) {
             p = new Parameter();
-            p.setName(name);
+            p.setName(searchParamCode);
             p.setValueString(humanName.getText().getValue());
             result.add(p);
         }
         return false;
     }
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, Identifier identifier) {
+        if (!TOKEN.equals(searchParamType)) {
+            throw invalidComboException(searchParamType, identifier);
+        }
         if (identifier != null && identifier.getValue() != null) {
```

Comment:
why don't do token check for Identifier?  https://www.hl7.org/fhir/search.html#token

Reply:
i don't understand the question

Reply:
ha, my mistake, the token check is exactly there